src/plugins/mf_priority.cpp

```diff
@@ -204,6 +204,71 @@ int check_queue_factor (flux_plugin_t *p,
  *                                                                            *
  *****************************************************************************/
 
+/*
+ * Get state of all user and bank information from plugin
+ */
+static int query_cb (flux_plugin_t *p,
+                     const char *topic,
+                     flux_plugin_arg_t *args,
+                     void *data)
+{
+    json_t *all_users = json_array (); // array of user/bank combos
+    json_t *user, *bank_info, *banks, *jtemp = NULL;
+
+    for (auto const &u : users) {
+        user = json_object (); // JSON object for one user
+        banks = json_array (); // array of banks that user belongs to
+
+        json_object_set_new (user, "userid", json_integer (u.first));
+
+        for (auto const &b : u.second) {
+            jtemp = NULL;
+            bank_info = json_object (); // JSON object for one bank
+
+            json_object_set_new (
+                bank_info, "bank", json_string (b.first.c_str ()));
+            json_object_set_new (
+                bank_info, "fairshare", json_real (b.second.fairshare));
+            json_object_set_new (
+                bank_info, "max_run_jobs", json_integer (
+                                                b.second.max_run_jobs));
+            json_object_set_new (
+                bank_info, "cur_run_jobs", json_integer (
+                                                b.second.cur_run_jobs));
+            json_object_set_new (
+                bank_info, "max_active_jobs", json_integer (
+                                                b.second.max_active_jobs));
+            json_object_set_new (
+                bank_info, "cur_active_jobs", json_integer (
+                                                b.second.cur_active_jobs));
```

Comment:
I was going to suggest that return codes be checked here, but maybe this would be simpler if you created `bank_info` with `json_pack()`? e.g.:

```c
    if (!(bank_info = json_pack ("{s:s s:f s:i s:i s:i s:i}",
                                 "bank", b.first.c_str (),
                                 "fairshare", b.second.fairshare,
                                 etc....    
```

Reply:
Ah, yeah, that might be simpler. I just pushed up a `fixup` commit that just uses `json_pack ()` as well as adds a `json_decref ()` to the JSON objects that are created. It definitely does reduce the number of lines needed to create the object. Let me know if that's what you had in mind (and/or if I didn't use the `decref ()` call correctly 😬)